Issue 188: Remove and update menuitems and menu mappings

scottaohara opened the pull request:
Closes #188 

removed element mappings for:
- a with parent menu
- button with parent menu
- input type=checkbox with parent menu
- input type=image with parent menu
- input type=radio with parent menu
- menuitem type=checkbox
- menuitem type=command
- menuitem type=radio

removed attribute mappings for:
- checked referencing menuitem types
- disabled referencing menuitem
- icon for menuitem element
- label referencing menu and menuitem elements
- radiogroup for menuitem element
- title for menuitem element

menu updated to map to role=list
  - add comment for info on role=menu to see ARIA


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/pull/214.html" title="Last updated on Sep 11, 2019, 1:29 PM UTC (6056938)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/214/24f3ea1...6056938.html" title="Last updated on Sep 11, 2019, 1:29 PM UTC (6056938)">Diff</a>

scottaohara committed:
remove commented out content

scottaohara committed:
remove deprecated HTML elements

menuitem type=radio, checkbox and command have been removed as they are not in the HTML spec

scottaohara committed:
remove menuitem element mappings

Remove additional elements that “mapped” to `menuitem` as this element is not in the living standard specification.

* `a` with parent `menu`
* `button` with parent `menu`
* `input type=checkbox` with parent `menu`
* `input type=image` with parent `menu`
* `input type=radio` with parent `menu`

scottaohara committed:
remove attributes referencing menuitem element

the `menuitem` element is not part of the Living Standard and the following attributes have been updated to remove references to `menuitem`.

* indeterminate idl referencing `menuitem`
* `checked` referencing `menuitem` types
* `disabled` referencing `menuitem`
* removed attr `icon` for `menuitem` element
* `label` referencing `menu` and `menuitem` elements
* `type` referencing `menuitem`
* removed attr `radiogroup` for `menuitem` element
* `title` for `menuitem` element

scottaohara committed:
update mapping for menu element

Per the HTML Living Standard, `<menu>` is a semantic alternative to `ul` to express an unordered list of commands.  There is (presently) no special mappings / specification definition for `li`s within a `menu`.

scottaohara committed:
cleanup and changelog

stevefaulkner commented on index.html:
```diff
@@ -2412,40 +2337,17 @@ <h3>HTML Element Role Mappings</h3>
+              <td class="atk"><div class="general">Use WAI-ARIA mapping</div></td>
+              <td class="ax"><div class="general">Use WAI-ARIA mapping</div></td>
+              <td class="comments">
+                The <a data-cite="HTML">`menu`</a> element is a semantic alternative to the <a data-cite="HTML">`ul`</a> element, and has no implemented mappings or behavior similar to ARIA's <a class="core-mapping" href="#role-map-menu">`role="menu"`</a>.
```
suggest rewording "has no implemented mappings or behavior similar to ARIA's <a class="core-mapping" href="#role-map-menu">`role="menu"`</a>"

to "has no implemented mappings or behavior that reflect the semantics of the ARIA <a class="core-mapping" href="#role-map-menu">`menu`</a> role.

scottaohara replied:
updated

scottaohara committed:
update menu comment per feedback

scottaohara committed:
Merge branch 'gh-pages' into issue_188

scottaohara committed:
Merge branch 'gh-pages' into issue_188

scottaohara commented:
Added "not ready for merge" label as there are some issues that need to be addressed outside of this PR before merging.

scottaohara commented:
Adding todo item of adding a link in the comment section for `menu` to point to HTML's obsolete section.

- [x] add https://html.spec.whatwg.org/multipage/obsolete.html#attr-menu-type

scottaohara committed:
add links to menu comments

link to HTML obsolete section for `menu type=` and `menuitem`

update date of change for changelog

scottaohara commented:
latest push completes todo to reference/link to obsolete `menuitem` and `menu type` in HTML spec.  changelog date updated.  this will also close #237 which is duplicate to this PR.
